eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/RetryExponential.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.eventhubs.implementation.ClientConstants;
+
+import java.time.Duration;
+
+public final class RetryExponential extends Retry {
+    private final Duration minBackoff;
+    private final Duration maxBackoff;
+    private final int maxRetryCount;
+    private final double retryFactor;
+
+    /**
+     * @param minBackoff The minimum time period permissible for backing off between retries.
+     * @param maxBackoff The maximum time period permissible for backing off between retries.
+     * @param maxRetryCount The maximum number of retries allowed.
+     */
+    public RetryExponential(Duration minBackoff, Duration maxBackoff, int maxRetryCount) {
+        this.minBackoff = minBackoff;
+        this.maxBackoff = maxBackoff;
+        this.maxRetryCount = maxRetryCount;
+        this.retryFactor = computeRetryFactor();
+    }
+
+    @Override
+    protected Duration onGetNextRetryInterval(final Exception lastException,
+                                              final Duration remainingTime,
+                                              final int baseWaitSeconds,
+                                              final int retryCount) {
+        if ((!Retry.isRetryableException(lastException)) || retryCount >= maxRetryCount) {
+            return null;
+        }
+        double nextRetryInterval = Math.pow(retryFactor, (double) retryCount);
```

Comment:
In general, if you know a variable won't be reassigned, I'd add `final` to it. This makes it explicit that you don't want this to change. So if, someone makes a mistake down the line and tries to reassign it, it'll result in a compiler error.